Common/Data/Consolidators/RangeConsolidator.cs

```diff
@@ -0,0 +1,223 @@
+        /// value is (x => x.Value) the <see cref="IBaseData.Value"/> property on <see cref="IBaseData"/></param>
+        /// <param name="volumeSelector">Extracts the volume from a data instance. The default value is null which does
+        /// not aggregate volume per bar, except if the input is a TradeBar.</param>
+        public RangeConsolidator(
+            decimal range,
+            Func<IBaseData, decimal> selector,
+            Func<IBaseData, decimal> volumeSelector = null)
+            : base(selector ?? (x => x.Value), volumeSelector ?? (x => x is TradeBar bar ? bar.Volume : 0))
+        {
+            Range = range;
+            _firstTick = true;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RangeConsolidator" /> class.
+        /// </summary>
+        /// <param name="range">The constant value size of each bar, where the unit is the minimum price variation of the RangeBar's symbol</param>
+        /// <param name="selector">Extracts the value from a data instance to be formed into a <see cref="RangeBar"/>. The default
+        /// value is (x => x.Value) the <see cref="IBaseData.Value"/> property on <see cref="IBaseData"/></param>
+        /// <param name="volumeSelector">Extracts the volume from a data instance. The default value is null which does
+        /// not aggregate volume per bar.</param>
+        public RangeConsolidator(decimal range,
+            PyObject selector,
+            PyObject volumeSelector = null)
+            : base(selector, volumeSelector)
+        {
+            Range = range;
+            _firstTick = true;
+        }
+
+        /// <summary>
+        /// Update the current RangeBar being created with the given data.
+        /// </summary>
+        /// <param name="time">Time of the given data</param>
+        /// <param name="currentValue">Value of the given data</param>
+        /// <param name="volume">Volume of the given data</param>
+        protected override void UpdateBar(DateTime time, decimal currentValue, decimal volume)
+        {
+            if (currentValue > CurrentRangeBar.Low + 2 * RangeSize)
+            {
+                Rising(time, currentValue, volume);
+            }
+            else if (currentValue < CurrentRangeBar.High - (2 * RangeSize))
+            {
+                Falling(time, currentValue, volume);
+            }
+            else
+            {
+                CurrentRangeBar.Update(time, currentValue, volume);
+            }
+        }
+
+        /// <summary>
+        /// Checks if the current RangeBar being created has closed. If that is the case, it consolidates
+        /// the bar, resets the current RangeBar and stores the needed information from the last bar to
+        /// create the next bar
+        /// <see cref="UpdateBar(DateTime, decimal, decimal)"/>
+        /// </summary>
+        protected override void CheckIfBarIsClosed()
```

Comment:
Maybe the name could be `ConsolidateIfBarIsClosed`. Just to make things clearer. A `Check..` method might be thougth of as something returning a boolean.